docs/src/idiomatic/assignments.md

```diff
@@ -0,0 +1,129 @@
+If, in the syntax order defined above, an expression containing a primed variable `x'` precedes an assignment to `x`, the assignment finder throws an exception of the following shape:
+```
+Assignment error: test.tla:10:16-10:17: x' is used before it is assigned.
+```
+notifying the user of any variables used before assignment. In particular, right-hand-sides of assignment candidates are subject to this restriction as well. Consider:
+```tla
+A == x' > 0 /\ x' = 1
+B == y' = x' + 2 /\ x' = 1
+```
+In `A`, the expression `x' > 0` precedes any assignment to `x` and in `B`, while `y' = x' + 2` is an assignment candidate for `y`, it precedes any assignment to `x`, so both expressions are inadmissible (and would trigger exceptions). 
+
+Note that this only holds true if `A` (resp. `B`) is chosen as the transition operator. If `A` is called inside another transition operator, for example in `Next == x' = 1 /\ A`, no exception is thrown.
+
+### Balance
+In cases of disjunction, both (resp. all) disjuncts must have assignments for the same set of variables. In particular, if one disjunct contains an assignment candidate and another does not, such as in this example:
+```tla
+/\ \/ y = 1
+   \/ y' = 2
+/\ y' = 3
+```
+the assignment finder will throw an exception of the following shape:
+```
+Assignment error: test.tla:10:15-10:19: Missing assignments to: y
+```
+notifying the user of any variables for which assignments exist in some, but not all, disjuncts. 
+Note that if we correct the above example to
+```tla
+/\ \/ y' = 1
+   \/ y' = 2
+/\ y' = 3
+```
+the assignments to `y` would be `y' = 1` and `y' = 2`, but not `y' = 3`; minimality prevents us from selecting all three, the syntax order constraint forces us to select assignments in `y' = 1 \/ y' = 2` before `y' = 3` and balance requires that we select both `y' = 1` and `y' = 2`.
+On the other hand, if we change the example to
+```tla
+/\ y' = 3
+/\ \/ y = 1
+   \/ y' = 2
+```
+the only assignment has to be `y' = 3`. While one of the disjuncts is an assignment candidate and the other is not, the balance requirement is not violated here, since neither disjunct is chosen as an assignment.
+
+Similar rules apply to if-then-else: both the `THEN _` and `ELSE _` branch must assign the same variables, however, the `IF _` condition is ignored when determining assignments.
+
+### Assignment-free Expressions
+Not all expressions may contain assignments. Given a transition operator `A`, based on the shape of `A`, the following holds:
+- If `A` has the shape `A_1 /\ ... /\ A_n`, then assignments are selected from `A_1, ... , A_n` sequentially, subject to the syntax-order rule.
+- If `A` has the shape `A_1 \/ ... \/ A_n`, then assignments are selected in all `A_1, ... , A_n` independently, subject to the balance rule.
+- If `A` has the shape `IF p THEN A_1 ELSE A_2`, then:
+    - `p` may not contain assignments. Any assignment candidates in `p` are subject to the assignment-before-use rule.
```

Comment:
This may be confusing. We still allow the expressions that look like assignments, but we do not recognize them as such. The current writing may read like you cannot have those expressions at all.

Reply:
I disagree, there is a defined and stressed difference between _assignments_ and _assignment candidates_. The text here in no way implies that assignment candidates are forbidden, in fact, it mentions them explicitly, e.g.
```
    - `p` may not contain assignments. Any assignment candidates in `p` are subject to the assignment-before-use rule.
```